### DISCUSSION TITLE
在 archlinux 上构建失败

### DISCUSSION THREAD

**Discussion by Luoyayu** [libnyanpasu/clash-nyanpasu#1392] [General]
构建命令 
`paru -S clash-nyanpasu-git --fm vim`
`pnpm i`
`pnpm check`
`pnpm prepare:nightly`
`pnpm build:nightly`

```
compiling reqwest v0.11.27
...
ompiling boa_utils v0.1.0 ($HOME/.cache/paru/clone/clash-nyanpasu-git/src/clash-nyanpasu/backend/boa_utils)
error[E0599]: no method named `bytes_stream` found for struct `reqwest::Response` in the current scope
   --> tauri/src/utils/downloader.rs:541:35
    |
541 |         let mut stream = response.bytes_stream();
    |                                   ^^^^^^^^^^^^ method not found in `Response`

error: could not compile `clash-nyanpasu` (bin "clash-nyanpasu") due to 1 previous error; 4 warnings emitted
       Error failed to build app: failed to build app

错误： 未能构建 'clash-nyanpasu-git-pre.release.r14.g521324da-2': 
错误： 未能构建的软件包：clash-nyanpasu-git-pre.release.r14.g521324da-2
```
[build-clash-nyanpasu-error.txt](https://github.com/user-attachments/files/16636309/build-clash-nyanpasu-error.txt)

**Comment by greenhat616** [comment]
使用的 rustc 版本为？
https://github.com/LibNyanpasu/clash-nyanpasu/actions/runs/10412550501/job/28838528104

**Reply by Luoyayu** [reply]
```
$ rustup toolchain list
stable-x86_64-unknown-linux-gnu (default)
nightly-x86_64-unknown-linux-gnu

$ rustc -V
rustc 1.80.1 (3f5fd8dd4 2024-08-06)
```

**Reply by keiko233** [reply]
> ```
> $ rustup toolchain list
> stable-x86_64-unknown-linux-gnu (default)
> nightly-x86_64-unknown-linux-gnu
> 
> $ rustc -V
> rustc 1.80.1 (3f5fd8dd4 2024-08-06)
> ```

请使用nightly

**Reply by Luoyayu** [reply]
![截图 2024-08-16 23-34-44](https://github.com/user-attachments/assets/050bf4f4-4c38-43b3-8b25-a9479177925d)
使用 `rust-nightly` 问题仍然存在（crates为官方源）

**Reply by greenhat616** [reply]
试一下将 tauri 目录下的 Cargo.toml 找到 reqwest 依赖，features 字段加上 stream 看看

**Comment by Luoyayu** [comment]
按如上更新 `backend/tauri/Cargo.toml` 后编译成功
